eo-maven-plugin/src/main/java/org/eolang/maven/DiscoverMojo.java

```diff
@@ -55,17 +54,19 @@
 public final class DiscoverMojo extends SafeMojo {
 
     @Override
-    public void exec() throws IOException {
+    public void exec() throws FileNotFoundException {
         final Collection<ForeignTojo> tojos = this.scopedTojos().notDiscovered();
         final Collection<String> discovered = new HashSet<>(1);
         for (final ForeignTojo tojo : tojos) {
             final Path src = tojo.optimized();
-            final Collection<String> names = this.discover(src);
-            for (final String name : names) {
-                this.scopedTojos().add(name).withDiscoveredAt(src);
-                discovered.add(name);
-            }
-            tojo.withDiscovered(names.size());
+            tojo.withDiscovered(
+                (int) this.discover(src)
+                    .stream()
+                    .filter(name -> !name.isEmpty())
+                    .peek(name -> this.scopedTojos().add(name).withDiscoveredAt(src))
+                    .peek(discovered::add)
```

Comment:
@volodya-lombrozo maybe we can join this line and line above into single one? Something like `.peek(name -> discovered.add(this.scopedTojos().add(name).withDiscoveredAt(src)))`

Reply:
@maxonfjvipon  `discovered.add(String)` accepts `String` object, `this.scopedTojos().add(name).withDiscoveredAt(src))` returns `ForeignTojo` object. So, it's impossible. Moreover, I don't see a problem in using `peek` several times.